### DISCUSSION TITLE
Questions regarding naming conventions and availability in C

### DISCUSSION THREAD

**Discussion by viswajith-g** [fraunhoferportugal/tsfel#138] [Q&A] [answered]
Hello, 

I am using TSFEL for a project and I have two questions:

1. Some of our features have names such as  '0_ECDF Percentile_0', ,0_ECDF Percentile Count_1', and '0_Wavelet energy_0'. I was trying to look for an explanation for the numbers 0 and 1 in this case. I believe it cannot be the sampling frequency because our SF is 50Hz and the documentation mentions the default sampling rate as 100Hz, but neither shows up in the feature names. Could somebody please explain what it could be?

2. Is there an equivalent C library for the Python package? I am trying to implement some feature extraction on an embedded device - specifically Nordic's nRF52840 and nRF52822 chips. I tried looking for some of them online and came up short, and was wondering if we would have to implement them in C by ourselves. 

I would appreciate it if anyone could answer these. 

Thanks,
Vish

**Comment by dmfolgado** [comment, marked as answer]
Hi there! Glad to know you're using tsfel.

1. Some features in our package return multiple values. When this happens, the numbers appended to the feature name indicate how many values the feature can return. Take the histogram feature as an example: it returns frequency values based on the number of bins used by default. If you use the default setting of 10 equal-width bins, then the values for that feature will span ten columns. Each column will have the feature name followed by a number indicating the bin.
We acknowledge the documentation could be clearer on this, and we'll work on improving it. Thank you for pointing it out!

2. Yes, we do have a separate C implementation for Nordic devices covering some of our feature set. I recommend reaching out to me directly to discuss this further.

**Reply by viswajith-g** [reply]
Thanks, sent an email!
